SimplenoteUITests/SimplenoteUITests.swift

```diff
@@ -0,0 +1,531 @@
+
+        Assert.labelExists(labelText: text_LoginEmailInvalid)
+        Assert.labelAbsent(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithNoPassword() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: "")
+
+        Assert.labelAbsent(labelText: text_LoginEmailInvalid)
+        Assert.labelExists(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithTooShortPassword() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataInvalidPassword)
+
+        Assert.labelAbsent(labelText: text_LoginEmailInvalid)
+        Assert.labelExists(labelText: text_LoginPasswordShort)
+    }
+
+    func testLogInWithExistingEmailIncorrectPassword() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataNotExistingPassword)
+        Assert.alertExistsAndClose(headingText: text_AlertHeading_Sorry, content: text_AlertContent_LoginFailed, buttonText: uidButton_Accept)
+    }
+
+    func testLogInWithCorrectCredentials() throws {
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataExistingPassword)
+        AllNotesAssert.screenShown()
+    }
+
+    func testLogOut() throws {
+        // Step 1
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataExistingPassword)
+        AllNotesAssert.screenShown()
+
+        // Step 2
+        _ = logOut()
+        Assert.signUpLogInScreenShown()
+    }
+
+    // Test below is automatically generated one, disabled for now to save time
+    /*
+    func testLaunchPerformance() throws {
+        if #available(macOS 10.15, iOS 13.0, tvOS 13.0, *) {
+            // This measures how long it takes to launch your application.
+            measure(metrics: [XCTApplicationLaunchMetric()]) {
+                XCUIApplication().launch()
+            }
+        }
+    }*/
+}
+
+class SimplenoteUISmokeTestsNoteEditor: XCTestCase {
+
+    let usualLinkText = "https://simplenote.com/"
+    let complexLinkPreviewText = "Simplenote"
+    let complexLinkRawText = "[Simplenote](https://simplenote.com/)"
+    let webViewTexts = [String](
+        arrayLiteral: "Simplenote",
+        "The simplest way to keep notes",
+        "All your notes, synced on all your devices. Get Simplenote now for iOS, Android, Mac, Windows, Linux, or in your browser.",
+        "Sign up now")
+
+    override class func setUp() {
+        app.launchArguments = ["enable-testing"]
+        app.launch()
+        let _ = attemptLogOut()
+        EmailLogin.open()
+        EmailLogin.logIn(email: testDataExistingEmail, password: testDataExistingPassword)
+        AllNotes.waitForLoad()
+    }
+
+    override func setUpWithError() throws {
+        continueAfterFailure = true
+        app.launch()
+
+        AllNotes.clearAllNotes()
+        Trash.empty()
+        AllNotes.open()
+    }
```

Comment:
There are two setup methods here. The first is the `class` setup, it will run only once, before the tests start; the second will run before every test.

Do we need them both? If so, I _think_ `continueAfterFailure` ought to be set in `class setUp` instead of being reset before every test. I also _think_ there's no need to run `app.launch()` again either. I say I _think_ because I haven't verified either, so correct me if I'm wrong, maybe I'm missing something 👌

Reply:
I'm using both classes because there's stuff that should be done only once for a pack of tests (e.g. log in under a certain account), and stuff that should be done before each test (like remove all notes and empty trash).

I tried moving `continueAfterFailure` to `setUp()` earlier, but this caused the error:

`Instance member 'continueAfterFailure' cannot be used on type 'SimplenoteUISmokeTestsNoteEditor'`

I did not look deep into it before, but decided to google a bit after your review, and [found](https://stackoverflow.com/a/62015300) that the error will not be shown if I change `override class func setUp()` to `override func setUp()`. 

As for removing `app.launch()` from `setUpWithError()`: I'd really like to do this, but about half of tests start failing after this. I need to look in detail why this happens, because this will save lots of execution time.